sushichef.py

```diff
@@ -73,17 +72,10 @@ def clean_video_title(title, lang_obj):
     return clean_title
 
 
-def generate_source_id(lang_code):
-    global SOURCE_ID_COUNTER
-    SOURCE_ID_COUNTER += 1
-    source_id = "arvind-{0}-{1}".format(lang_code, SOURCE_ID_COUNTER)
-    return source_id
-
-
-def include_video_topic(topic_node, video_data, lang_obj):
+def include_video_topic(topic_node, video_data, lang_obj, counter):
     # Include video details to the parent topic node
-    video = video_data
-    video_source_id = generate_source_id(lang_obj.code)
+    video_id = video.uid
+    video_source_id = 'arvind-{0}-video-topic-{1}-{2}'.format(lang_obj.name, video_id, counter)
```

Comment:
We're creating a VideoNode and a YouTubeVideoFile, meaning this code handles a specific video, and so video.uid should be specific to that video. Why are we using a counter here?

Reply:
I already removed it in my recent changes @kollivier